### PR TITLE
High cardinality

### PR DESCRIPTION
**Pull request by ashwinkumar12345**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/ad/index.md [hunk 1]
```diff
@@ -55,12 +55,39 @@ For example, if you choose `min()`, the detector focuses on finding anomalies ba
...
 {: .note }
 
-1. On the **Features** page, select **Add features**.
-1. Enter the **Name** of the feature.
+1. On the **Model configuration** page, enter the **Feature name**.
 1. For **Find anomalies based on**, choose the method to find anomalies. For **Field Value** menu, choose the **field** and the **aggregation method**. Or choose **Custom expression**, and add in your own JSON aggregation query.
-1. Preview sample anomalies and adjust the feature settings if needed.
-- For sample previews, the anomaly detection plugin selects a small number of data samples---for example, one data point every 30 minutes---and uses interpolation to estimate the remaining data points to approximate the actual feature data. It loads this sample dataset into the detector. The detector uses this sample dataset to generate a sample preview of anomaly results.
+
+#### (Optional) Set a category field
```
Comment (sean-zheng-amazon) [inline docs/ad/index.md:61]: can we call out that this is for the support of high cardinality?

File: docs/ad/index.md [hunk 2]
```diff
@@ -55,12 +55,39 @@ For example, if you choose `min()`, the detector focuses on finding anomalies ba
...
+You can categorize anomalies based on a keyword or IP field type.
+
+If you specify a category in the same time series but sliced with a different dimension like IP addresses, product IDs, country codes, and so on, you’ll see a granular view of anomalies within each entity of that field. This helps to dive deeper into anomalies of a unique entity or ID and isolate and debug issues.
+
+To set a category field, choose **Enable a category field** and select a field.
+
+Only a certain number of unique entities are supported in the category field. Use the following equation to calculate the recommended total number of entities number supported in a cluster:
+
+```
+(JvmHeapSizeInMb / 20) * (DataNodesCount)
```
Comment (sean-zheng-amazon) [inline docs/ad/index.md:72]: @weicongs-amazon @kaituo, can you guys comment on this formula? does this mean each entity takes 20M memory?

File: docs/ad/index.md [hunk 3]
```diff
@@ -86,6 +113,11 @@ If you see the detector pending in "initialization" for longer than a day, aggre
 
 Anomaly grade is a number between 0 and 1 that indicates the level of severity of how anomalous a data point is. An anomaly grade of 0 represents “not an anomaly,” and a non-zero value represents the relative severity of the anomaly. The confidence score is an estimate of the probability that the reported anomaly grade matches the expected anomaly grade. Confidence increases as the model observes more data and learns the data behavior and trends. Note that confidence is distinct from model accuracy.
 
+If you set the category field, you see an additional **Heat map** chart. The heat map correlates results for anomalous entities.
```
Comment (sean-zheng-amazon) [inline docs/ad/index.md:116]: one feedback we recieved from uxdr is that the initial status of the result / feature plots is empty, which might be confusing for users, can we add a notes here that the plots will be empty till users select the anomaly entity?

**Review by sean-zheng-amazon — COMMENTED** [review]
overall looks good, please see comments inline.

File: docs/ad/index.md [hunk 4]
```diff
@@ -55,12 +55,39 @@ For example, if you choose `min()`, the detector focuses on finding anomalies ba
...
+(JvmHeapSizeInMb / 20) * (DataNodesCount)
+```
+
+For example, for a cluster with 3 data nodes, each with 8G of JVM heap size, the total number of unique entities supported is (8096 / 20 ) * 3 = 1200.  
+
+#### Set a window size
+
+Set the number of aggregation intervals from your data stream to consider in a detection window. We recommend you choose this value based on your actual data to see which one leads to the best results for your use case.
+
+Based on experiments performed on a wide variety of one-dimensional data streams, we recommend using a window size between 1 and 16. The default window size is 8.
```
[the remark this answers is not in the document]
Comment (ylwu-amzn) [reply docs/ad/index.md:81]: Sorry, seems we have changed to 1, but there is a TODO saying will change to 4, https://github.com/opendistro-for-elasticsearch/anomaly-detection/blob/5093c8a4df09fe665fe93eb6e5a994206adcfc2b/src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java#L233

File: docs/ad/index.md (hunk 3 above)
Comment (ylwu-amzn) [reply docs/ad/index.md:116]: And for high cardinality detector, we only show anomaly and feature line chart for time period which has anomaly (anomaly_grade > 0).

File: docs/ad/index.md (hunk 1 above)
Comment (ashwinkumar12345) [reply docs/ad/index.md:61]: added

File: docs/ad/index.md (hunk 3 above)
Comment (ashwinkumar12345) [reply docs/ad/index.md:116]: added

File: docs/ad/index.md [hunk 5]
```diff
@@ -55,12 +55,39 @@ For example, if you choose `min()`, the detector focuses on finding anomalies ba
...
+1. On the **Model configuration** page, enter the **Feature name**.
 1. For **Find anomalies based on**, choose the method to find anomalies. For **Field Value** menu, choose the **field** and the **aggregation method**. Or choose **Custom expression**, and add in your own JSON aggregation query.
-1. Preview sample anomalies and adjust the feature settings if needed.
-- For sample previews, the anomaly detection plugin selects a small number of data samples---for example, one data point every 30 minutes---and uses interpolation to estimate the remaining data points to approximate the actual feature data. It loads this sample dataset into the detector. The detector uses this sample dataset to generate a sample preview of anomaly results.
+
+#### (Optional) Set a category field
+
+You can categorize anomalies based on a keyword or IP field type.
+
+If you specify a category in the same time series but sliced with a different dimension like IP addresses, product IDs, country codes, and so on, you’ll see a granular view of anomalies within each entity of that field. This helps to dive deeper into anomalies of a unique entity or ID and isolate and debug issues.
```
[the remark this answers is not in the document]
Comment (ashwinkumar12345) [reply docs/ad/index.md:65]: modified. please review.

File: docs/ad/index.md (hunk 2 above)
Comment (kaituo) [reply docs/ad/index.md:72]: I thought we cannot have a formula due to the query size limit. 

Ignoring the query size limit, this is the formula:

```
# data nodes * heap size * AD memory max percentage / entity size of an detector
```

AD memory max percentage is 10% by default (adjustable via ES setting).  We can fetch entity size of an detector using profile API.

For example, for a cluster with 3 data nodes, each with 8G of JVM heap size, AD memory max percentage is 10% (default value) , entity size of the detector is 1MB,  the total number of unique entities supported is (8.096  * 10^ 9 *0.1 / 1M  ) * 3 = 2429

File: docs/ad/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kaituo) [reply docs/ad/index.md:81]: yeah, high cardinality detector's shingle size is 1 now.

File: docs/ad/index.md (hunk 2 above)
Comment (ashwinkumar12345) [reply docs/ad/index.md:72]: Added to the docs. Please review.

**Pull request merged**